apps/src/componentLibrary/typography/SpecificComponentsGenerator.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import Typography from './index';
+import {
+  SemanticTag,
+  VisualApproach,
+  SpecificTypographyElementProps
+} from './types';
+
+interface SpecificTypographyElementToGenerateTemplate {
+  displayName: string;
+  semanticTag: SemanticTag;
+  defaultVisualApproach?: VisualApproach;
+}
+const specificTypographyComponentsToGenerate: SpecificTypographyElementToGenerateTemplate[] =
+  [
+    {displayName: 'Heading1', semanticTag: 'h1'},
+    {displayName: 'Heading2', semanticTag: 'h2'},
+    {displayName: 'Heading3', semanticTag: 'h3'},
+    {displayName: 'Heading4', semanticTag: 'h4'},
+    {displayName: 'Heading5', semanticTag: 'h5'},
+    {displayName: 'Heading6', semanticTag: 'h6'},
+    {
+      displayName: 'BodyOneText',
+      semanticTag: 'p',
+      defaultVisualApproach: 'body-one'
+    },
+    {
+      displayName: 'BodyTwoText',
+      semanticTag: 'p',
+      defaultVisualApproach: 'body-two'
+    },
+    {
+      displayName: 'OverlineText',
+      semanticTag: 'p',
+      defaultVisualApproach: 'overline'
+    },
+    {displayName: 'EmText', semanticTag: 'em'},
+    {displayName: 'StrongText', semanticTag: 'strong'},
+    {displayName: 'Figcaption', semanticTag: 'figcaption'}
+  ];
+
+const generateSpecificTypographyComponents = (
+  componentsToGenerate: SpecificTypographyElementToGenerateTemplate[]
+): {[key: string]: React.FunctionComponent<SpecificTypographyElementProps>} =>
+  componentsToGenerate.reduce((acc, componentTemplateData) => {
+    const {displayName, semanticTag, defaultVisualApproach} =
+      componentTemplateData;
+
+    const SpecificTypographyComponent: React.FunctionComponent<
+      SpecificTypographyElementProps
+    > = componentProps => {
+      const {visualApproach, children, className, style} = componentProps;
+
+      return (
+        <Typography
+          semanticTag={semanticTag}
+          visualApproach={visualApproach || defaultVisualApproach}
+          className={className}
+          style={style}
+        >
+          {children}
+        </Typography>
+      );
+    };
+    SpecificTypographyComponent.displayName = displayName;
```

Comment:
Curious - why do we need to assign displayName here?

Reply:
Good question, thank you!
We meed display to improve dev experience, especially in react-dev-tools. If we won't specify the displayName here it will be display by the name of the variable it was first decalred - meaning all of the Typography Elements will be called SpecificTypographyComponent.
Here's a couple of screenshots of the same component in the react-dev-tools (with and without displayName specified).
<img width="526" alt="Знімок екрана 2023-04-21 о 10 22 09" src="https://user-images.githubusercontent.com/22244040/233570630-d845b861-5762-4583-b9a5-5e6f52e53c4c.png">
<img width="526" alt="Знімок екрана 2023-04-21 о 10 23 48" src="https://user-images.githubusercontent.com/22244040/233570655-84031b8f-44b0-4e60-b514-5930b172fff9.png">